static/app/components/events/eventTags/eventTagsTree.tsx

```diff
@@ -0,0 +1,202 @@
+import styled from '@emotion/styled';
+
+import EventTagsContent from 'sentry/components/events/eventTags/eventTagContent';
+import {space} from 'sentry/styles/space';
+import type {EventTag} from 'sentry/types';
+import {generateQueryWithTag} from 'sentry/utils';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+
+const MAX_TREE_DEPTH = 4;
+interface TagTree {
+  [key: string]: TagTreeContent;
+}
+interface TagTreeContent {
+  subtree: TagTree;
+  value: string;
+  // These will be omitted on pseudo tags (see addToTagTree)
+  meta?: Record<any, any>;
+  originalTag?: EventTag;
+}
+
+function addToTagTree(
+  tree: TagTree,
+  tag: EventTag,
+  meta: Record<any, any>,
+  originalTag: EventTag
+): TagTree {
+  const branchMatches = tag.key.match(/\./g) ?? [];
```

Comment:
do these regexs need to be global? if we can make them not globals, the regexs should be pulled out of the hot path into constants outside the function
```suggestion
  const branchMatches = tag.key.match(dotRegex) ?? [];
```